formats/cbor/commonMain/src/kotlinx/serialization/cbor/CborDecoder.kt

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.cbor
+
+public interface CborDecoder {
+    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
+    public val cbor: Cbor
```

Comment:
Have you tried to implement `ByteStringWrapper` separately? Because I've just realized that all configuration flags in `Cbor` are `internal val`, so it would be of little use to third-party clients. We either need to create a separate class `CborConfiguration` (by analogue with Json) or just make them `public`.

Reply:
Ah, we do not need configuration flags for wrapper, we need only `cbor.encodeToByteArray`. Do we need configuration for other tasks though?

Reply:
I don't think we can go wrong with making those flags publicly accessible. If someone wants to implement cbor-specific serializers, I'd assume the want access to these flags. Besides, those are read-only anways so I don't see any issue with making them public

Reply:
made all flags public in 62b80b56188e9ba65c10ff513fbcb3e1607d9603